apps/condo/domains/resident/tasks/helpers/sendBillingReceiptsAddedNotifications.js

```diff
@@ -95,27 +99,25 @@ const prepareAndSendNotification = async (context, receipt, resident) => {
  */
 const sendBillingReceiptsAddedNotificationsForPeriod = async (receiptsWhere, onLastDtChange) => {
     const { keystone: context } = await getSchemaCtx('Resident')
-    const receiptsCount = await BillingReceipt.count(context, receiptsWhere)
     let skip = 0, successCount = 0
     const notifiedUsers = new Set()
 
-    logger.info({ msg: 'sending billing receipts', receiptsCount, data: receiptsWhere })
-
-    // Exit if no receipts found to proceed
-    if (!receiptsCount) return
-
-    let lastReceipt
-
-    while (skip < receiptsCount) {
-        const receipts = await BillingReceipt.getAll(context, receiptsWhere, { sortBy: ['createdAt_ASC'], first: CHUNK_SIZE, skip })
+    let lastReceipt, currBatchLength = CHUNK_SIZE
 
+    while (currBatchLength === CHUNK_SIZE) {
```

Comment:
we already have loadListByChunks, why implementing batching yourself?

Reply:
I think that it would be best to just create function that allows you to do smth with the batch and use it here

Reply:
if I use the loadListByChank utility, it means that I must first receive all suitable receipts and then run the push notification logic. In this case, I will store a huge number of receipts in memory, which will greatly affect the memory consumption. Receipts are the heaviest model in our CRM if I'm not mistaken.

Reply:
Now the task processes the chunk, then removes it from the map and goes for the next one.

Reply:
Yes, but maybe lets implement smth like loadAndProcess(..., processChunk: () => {})

Reply:
Update, you should not use loadListByChunks, your only option is to use `find` check `ADR-36` for more details